### PR TITLE
[2.29] Show name of option when downloading Event report Excel

### PR DESCRIPTION
**Pull request by luciano-fiandesio**
- DHIS2-2743
- Display the actual NAME of the option rather than the CODE in the
Excel Event report

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java [hunk 1]
```diff
@@ -694,4 +699,38 @@ private static WritableWorkbook openWorkbook( OutputStream outputStream )
 
         return Workbook.createWorkbook( outputStream, ws );
     }
+
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java]: Move to AnalyticsUtils

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java [hunk 2]
```diff
@@ -694,4 +699,38 @@ private static WritableWorkbook openWorkbook( OutputStream outputStream )
...
+    {
+        Map<String, String> metadataCache = new HashMap<>();
+        Map items = (Map) grid.getMetaData().get( "items" );
+
+        for ( Object key : items.keySet() )
+        {
+            Object metadataItem = items.get( key );
+            try {
+                metadataCache.put( (String) key, getMetadataItemName( metadataItem ) );
+            } catch (IllegalAccessException | InvocationTargetException e) {
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java]: Codestyle - new line

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java [hunk 3]
```diff
@@ -694,4 +699,38 @@ private static WritableWorkbook openWorkbook( OutputStream outputStream )
...
+                // unable to create cache entry
+                log.warn("Unable to add a cache entry from a metadata item.", e);
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java]: Code style - spaces after ( )

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java [hunk 4]
```diff
@@ -694,4 +699,38 @@ private static WritableWorkbook openWorkbook( OutputStream outputStream )
...
     }
+    
+    private static Map<String, String> buildMetadataCache( Grid grid )
+    {
+        Map<String, String> metadataCache = new HashMap<>();
+        Map items = (Map) grid.getMetaData().get( "items" );
+
+        for ( Object key : items.keySet() )
+        {
+            Object metadataItem = items.get( key );
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java]: Cast to IdentifiableObject

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java [hunk 5]
```diff
@@ -871,6 +875,24 @@ public Grid substituteMetaData( int sourceColumnIndex, int targetColumnIndex, Ma
...
+    public Grid substituteValues( Map<String, String> uidToNameMap )
+    {
+        for ( List<Object> columns : grid )
+        {
+            for ( int col = 0; col < columns.size(); col++ )
+            {
+                Object column = columns.get( col );
+                if ( column != null && !MathUtils.isNumeric( String.valueOf( column ) ) )
+                {
+                    columns.set( col, uidToNameMap.getOrDefault( String.valueOf( column ), String.valueOf( column ) ) );
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java]: Regarding 

`uidToNameMap.getOrDefault( String.valueOf( column ), String.valueOf( column ) )`

is this intentional? is the default the same as the initial value?

Comment (luciano-fiandesio) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java]: Fixed

**Pull request merged**